examples/WiFiPing/WiFiPing.ino

```diff
@@ -0,0 +1,136 @@
+/*
+
+ This example connects to a encrypted Wifi network (WPA/WPA2).
+ Then it prints the  MAC address of the Wifi shield,
+ the IP address obtained, and other network details.
+ Then it continuously pings given IP Address.
+
+ Circuit:
+ * WiFi shield attached / MKR1000
+
+ created 13 July 2010
+ by dlf (Metodo2 srl)
+ modified 13 May 2016
+ by Petar Georgiev
+ */
+#include <SPI.h>
+#include <WiFi101.h>
+
+char ssid[] = "yourNetwork";     //  your network SSID (name)
+char pass[] = "secretPassword";  // your network password
+int status = WL_IDLE_STATUS;     // the Wifi radio's status
+
+String hostName = "www.google.com";
+byte pingResult;
+
+void setup() {
+  //Initialize serial and wait for port to open:
+  Serial.begin(9600);
+  while (!Serial) {
+    ; // wait for serial port to connect. Needed for native USB port only
+  }
+
+  // check for the presence of the shield:
+  if (WiFi.status() == WL_NO_SHIELD) {
+    Serial.println("WiFi shield not present");
+    // don't continue:
+    while (true);
+  }
+
+  // attempt to connect to Wifi network:
+  while ( status != WL_CONNECTED) {
+    Serial.print("Attempting to connect to WPA SSID: ");
+    Serial.println(ssid);
+    // Connect to WPA/WPA2 network:
+    status = WiFi.begin(ssid, pass);
+
+    // wait 5 seconds for connection:
+    delay(5000);
+  }
+
+  // you're connected now, so print out the data:
+  Serial.print("You're connected to the network");
+  printCurrentNet();
+  printWifiData();
+
+}
+
+void loop() {
+
+  Serial.print("Pinging ");
+  Serial.print(hostName);
+  Serial.print(": ");
+  
+  pingResult = WiFi.ping(hostName);
+  
+  if (pingResult == WL_PING_SUCCESS){
+    Serial.println("SUCCESS");
+  } else {
+    switch (pingResult){
+      case WL_PING_DEST_UNREACHABLE: { Serial.println("Destination host unreachable"); }; break;
+      case WL_PING_TIMEOUT: { Serial.println("Request Timed Out"); }; break;
+      case WL_PING_UNKNOWN_HOST: { Serial.println("Unable to resolve hostname to IP Address"); }; break;
+      case WL_PING_ERROR: { Serial.println("Internal error"); }; break;  
+    };      
+  };

```

Comment:
It might be easier for beginners to understand if the switch was changed to ```if/else if``` statements. I don't remember seeing any examples that use a switch statement.